src/protobuf-net.BuildTools/Generators/ProtoFileGenerator.cs

```diff
@@ -113,6 +113,11 @@ string ShowVersion(Version? version)
                     log?.Invoke($"Processing '{name}' relative to '{location}'");
 
                     var userOptions = context.AnalyzerConfigOptions.GetOptions(schema.Value);
+                    if (userOptions.TryGetValue(Literals.AdditionalFileMetadataPrefix + "SuppressOutput", out var optionValue) && bool.TryParse(optionValue, out var suppressOutput) && suppressOutput)
```

Comment:
I wonder if this disables too much - idea: what if we do all the usual processing and adding, so the file is fully processed and available, but we set `IncludeInOutput` to `false`? This property already exists, and is IIRC optionally available during `Add` (or can be set afterwards per file). This also makes me think maybe naming `IncludeInOutput` might be clearer, and default to true? Thoughts?

Reply:
FWIW a lot of similar things I've seen were "IncludeInProject" in case that idea helps (I may be misunderstanding what "output" means in this context/scenario)

Reply:
In this case, it is "included" (for the purposes of other files to cross-reference) either way; it is specifically the output part that changes.

Reply:
I have no problem with this.

But I must admit, I'm not knowledgeable enough in the code base to know the benefits and/or drawbacks of processing the file even though it isn't included in the output.

Here's a question though. You said available, what does that mean in this context? Does that mean the imported files type definitions are available in some global parsing scope?

Reply:
Well, if we didn't want it to be available: *why are we defining it as an additional file in the first place*? If it ends up getting imported, it would be parsed anyway - this just forces parse/validate, which seems fine to me, since they've included it in the model.

Reply:
Sounds reasonable. I've updated the PR.